components/service/glean/src/main/java/mozilla/components/service/glean/storages/TimespansStorageEngine.kt

```diff
@@ -92,42 +93,14 @@ internal open class TimespansStorageEngineImplementation(
         userPreferences?.putString(storeName, tuple.toString())
     }
 
-    /**
-     * Add the elapsed time to the time currently stored in the metric.
-     *
-     * @param metricData the metric information for the timespan
-     * @param timeUnit the time unit we want the data in when snapshotting
-     * @param elapsedNanos the time to record, in nanoseconds
-     */
-    @Synchronized
-    fun sum(
-        metricData: CommonMetricData,
-        timeUnit: TimeUnit,
-        elapsedNanos: Long
-    ) {
-        // Look for the start time: if it's there, commit the timespan.
-        val timespanName = metricData.identifier
-
-        // Store the time unit: we'll need it when snapshotting.
-        timeUnitsMap[timespanName] = timeUnit
-
-        // Use a custom combiner to sum the new timespan to the one already stored. We
-        // can't adjust the time unit before storing so that we still allow for values
-        // lower than the desired time unit to accumulate.
-        super.recordMetric(metricData, elapsedNanos, timeUnit) { currentValue, newValue ->
-            currentValue?.let {
-                it + newValue
-            } ?: newValue
-        }
-    }
-
     /**
      * Set the elapsed time explicitly.
      *
      * @param metricData the metric information for the timespan
      * @param timeUnit the time unit we want the data in when snapshotting
      * @param elapsedNanos the time to record, in nanoseconds
      */
+    @Suppress("LongMethod")
```

Comment:
This is starting to get ridiculous... (that's not about you or your code, but about our linters)

Reply:
Yes :\ I agree :\